targets/TARGET_ARM_SSG/TARGET_CM3DS_MPS2/gpio_api.c

```diff
@@ -0,0 +1,153 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "gpio_api.h"
+#include "pinmap.h"
+
+#define GPIO_NUM    5
+
+CMSDK_GPIO_TypeDef* GPIO_MAP[GPIO_NUM] = {
+    CMSDK_GPIO0,
+    CMSDK_GPIO1,
+    CMSDK_GPIO2,
+    CMSDK_GPIO3,
+    CMSDK_GPIO4
+};
+
+// function to enable the GPIO pin
+uint32_t gpio_set(PinName pin) {
+    return (1);
+}
+
+//function to initialise the gpio pin
+// this links the board control bits for each pin
+// with the object created for the pin
+void gpio_init(gpio_t *obj, PinName pin) {
+        if(pin == NC){ return;}
+        else {
+        int pin_value = 0;
+        obj->pin = pin;
+        if(pin <=15){
+            pin_value = pin;
+        }else if (pin >= 16 && pin <= 31){
+            pin_value = pin-16;
+        }else if (pin >= 32 && pin <= 47){
+            pin_value = pin-32;
+        }else if (pin >= 48 && pin <= 51){
+            pin_value = pin-48;
+        }else if (pin == 100 || pin == 101){
+            pin_value = pin-100;
+        }else if (pin == 110 || pin == 111){
+            pin_value = pin-110;
+        }else if (pin >= 200 && pin <= 207){
+            pin_value = pin-200;
+        }else if (pin >= 210 && pin <= 217){
+            pin_value = pin-210;
+        }else if (pin == 303){
+            pin_value = pin-302;
+        }else if (pin == 307){
+            pin_value = pin-307;
+        }else if (pin == 308){
+            pin_value = pin-305;
+        }else if (pin == 309){
+            pin_value = pin-305;
+        }else if (pin == 310){
+            pin_value = pin-305;
+        }else if (pin == 311){
+            pin_value = pin-305;
```

Comment:
These pin_values are calculated to match the bits in in the FPGAIO->MISC register
  __IO uint32_t MISC;            // Offset: 0x04C (R/W)  Misc control */
                                 //                         [31:10] : Reserved
                                 //                            [9] : SHIELD_1_SPI_nCS
                                 //                            [8] : SHIELD_0_SPI_nCS
                                 //                            [7] : ADC_SPI_nCS
                                 //                            [6] : CLCD_BL_CTRL
                                 //                            [5] : CLCD_RD
                                 //                            [4] : CLCD_RS
                                 //                            [3] : CLCD_RESET
                                 //                            [2] : RESERVED
                                 //                            [1] : SPI_nSS
                                 //                            [0] : CLCD_CS